components/merkledb/src/proof_map_index/proof.rs

```diff
@@ -440,6 +440,27 @@ impl<K, V> MapProof<K, V> {
         debug_assert!(self.proof.windows(2).all(|w| w[0].path < w[1].path));
         self
     }
+
+    /// Creates MapProof from provided `proof` and `entries` vectors. Used to construct proof
+    /// after deserialization.
+    pub fn from_raw_parts(proof: &[(ProofPath, Hash)], entries: Vec<(K, Option<V>)>) -> Self {
```

Comment:
I think there is trade off, because to implement `ProtobufConvert` for `MapProof` in `merkledb` crate we need to add proto files and `exonum-build` dependency(to use exonum protobuf types).  
In one hand we need to hide inner proof details, in other hand we don't want to add extra dependencies because IMO `merkledb` should not be depended on protobuf.

Reply:
Extracting `ProtobufConvert` is orthogonal to the question where to implement `ProtobufConvert` for proofs 🙃 Also, I'd imagine if `ProtobufConvert` *is* extracted, the types that proofs depend on would have `ProtobufConvert` implementations directly in the `exonum-crypto` crate (maybe, gated via a feature, since having Protobuf compiler locally installed is a non-trivial complication, which is unnecessary in some scenarios). Likewise, proof implementations could be gated via a feature.

In general, I don't think that implementing Protobuf (de)serialization is significantly different from implementing the `serde` one.

Reply:
> where to implement ProtobufConvert for proofs

Due to orphan rules, it's possible only in MerkleDB, isn't it?

Reply:
...Or in the crate defining `ProtobufConvert`, as it is done in the PR currently.